t/ui/21-admin-needles.t

```diff
@@ -138,6 +139,20 @@ subtest 'dereference symlink when displaying needles info' => sub {
             last_matched_module_id => undef,
             file_present           => 1,
         });
+    push @need_deleted_needles, $symlink_needle;
+    for (my $i = 1; $i < 10; $i++) {
```

Comment:
the "10" is a magic number that you now need to repeat multiple times. Why are 10 needed? I suggest we design a test with the minimal amount of data that we need to show a certain feature. Is 2 enough?

Reply:
Because the web page loads 10 needles at one time.  I need to make sure the needles in the first page are all symlink.

Reply:
I see. Maybe better if you can read out the number of items per page instead of hardcoding? Otherwise please use a constant with a descriptive name and the value 10